pyscriptjs/tests/integration/support.py

```diff
@@ -219,27 +219,34 @@ def goto(self, path):
         url = f"{self.http_server}/{path}"
         self.page.goto(url, timeout=0)
 
-    def wait_for_console(self, text, *, timeout=None, check_js_errors=True):
+    def wait_for_console(self, text, *, timeout=30 * 1000, check_js_errors=True):
         """
-        Wait until the given message appear in the console.
+        Wait until the given message appear in the console. If the message was
+        already printed in the console, return immediately.
 
         Note: it must be the *exact* string as printed by e.g. console.log.
-        If you need more control on the predicate (e.g. if you want to match a
-        substring), use self.page.expect_console_message directly.
 
         timeout is expressed in milliseconds. If it's None, it will use
         playwright's own default value, which is 30 seconds).
 
         If check_js_errors is True (the default), it also checks that no JS
         errors were raised during the waiting.
         """
-
-        def pred(msg):
-            return msg.text == text
-
+        # NOTE: we cannot use playwright's own page.expect_console_message(),
+        # because if you call it AFTER the text has already been emitted, it
+        # waits forever. Instead, we have to use our own custom logic.
         try:
-            with self.page.expect_console_message(pred, timeout=timeout):
-                pass
+            t0 = time.time()
+            while True:
+                elapsed_ms = (time.time() - t0) * 1000
+                if elapsed_ms > timeout:
+                    raise TimeoutError()
+                #
+                if text in self.console.all.lines:
+                    # found it!
+                    return
+                #
+                self.page.wait_for_timeout(50)
```

Comment:
What about something like:
```
if text in self.console.all.lines:
    return
with self.page.expect_console_message(pred, timeout=timeout):
    pass
```
there would still be a race condition where the message appears between the two checks, but it would be much less likely.

... So probably what you did is better.

Reply:
I am not sure what to answer 😅
The reason why I didn't do what you describe is precisely because there is a race condition, so I don't think we should switch.

What I *could* do is to swap the `if timeout` and `if text` blocks, if you think it makes the code more readable

Reply:
I switched the two blocks in 4dea44f because it actually makes the code more readable